serde/src/private/de.rs

```diff
@@ -1113,9 +1115,13 @@ mod content {
         V: Visitor<'de>,
         E: de::Error,
     {
-        let map = content
-            .into_iter()
-            .map(|(k, v)| (ContentDeserializer::new(k), ContentDeserializer::new(v)));
+        fn content_deserializer_pair<'de, E>(
+            (k, v): (Content<'de>, Content<'de>),
+        ) -> (ContentDeserializer<'de, E>, ContentDeserializer<'de, E>) {
+            (ContentDeserializer::new(k), ContentDeserializer::new(v))
+        }
+
+        let map = content.into_iter().map(content_deserializer_pair);
```

Comment:
You also can just pass `content.into_iter()` because `Content` implements `IntoDeserializer` which is used in `MapAccess` implementation:
https://github.com/serde-rs/serde/blob/c82f2580b43c5d44f557bf530554081e6fed646d/serde/src/de/value.rs#L1200-L1207

The same implicit conversion can be applied to use of `SeqDeserializer` and, if implement `IntoDeserializer for &Content`, for `ContentRefDeserializer`.

Maybe this also could reduce generated size?